src/SleighAsm.h

```diff
@@ -0,0 +1,70 @@
+#ifndef R2GHIDRA_SLEIGHASM_H
+#define R2GHIDRA_SLEIGHASM_H
+
+#include <string>
+#include <vector>
+#include <r_core.h>
+#include "architecture.hh"
+#include "sleigh_arch.hh"
+
+class AsmLoadImage : public LoadImage
+{
+    private: 
+        RIO *io = nullptr;
+
+    public: 
+        AsmLoadImage(RIO *io) : LoadImage("radare2_program"), io(io) {}
+        virtual void loadFill(uint1 *ptr,int4 size,const Address &addr)
+        {
+            r_io_read_at(io, addr.getOffset(), ptr, size);
+        }
+        virtual string getArchType(void) const { return "radare2"; }
+        virtual void adjustVma(long adjust)
+        {
+            throw LowlevelError("Cannot adjust radare2 virtual memory");
+        }
+};
+
+class AssemblySlg : public AssemblyEmit
+{
+	public:
+        char *str = nullptr;
+
+		void dump(const Address &addr, const string &mnem, const string &body) override
+		{
+			str = r_str_newf("%s %s", mnem.c_str(), body.c_str());
+		}
+
+        ~AssemblySlg() { if (str) free (str); }
+};
+
+class SleighAsm
+{
+    private:
+        AsmLoadImage loader;
+        Sleigh trans;
+        ContextInternal context;
+        DocumentStorage docstorage;
+        std::string sleigh_id;
+        FileManage specpaths;
+        std::vector<LanguageDescription> description;
+        int languageindex;
+        int alignment;
+
+        RConfig *get_config(RAsm *a);
```

Comment:
Please use `lowerCamelCase` for all method names. For functions use `UpperCamelCase` and for variables use `snake_case`.